docs/server-push.md

```diff
@@ -0,0 +1,34 @@
+
+Starlette includes support for HTTP/2 and HTTP/3 server push, making it
+possible to push resources to the client to speed up page load times.
```

Comment:
We probably want to note that `hypercorn` is currently the only ASGI server to support server push.

Reply:
That's not *strictly* true, aioquic's demo server also supports server push.

Also if you're open to the idea I'd be happy to submit a PR to uvicorn

Reply:
> Also if you're open to the idea I'd be happy to submit a PR to uvicorn

Absolutely, yup!